components/Users/UserModal/userModal.tsx

```diff
@@ -25,13 +25,61 @@ const UserModal: React.FunctionComponent<UserModalProps> = ({
   isFamilyPath,
   setIsOpenFam,
 }: UserModalProps) => {
-  const [allTransactions, setTransactions] = useState([]);
+  const [isEditing, setIsEditing] = useState(false);
+  const [currUser, setCurrUser] = useState(user);
+  const [role, setRole] = useState(user?.family_head ? "Head" : "Member");
+  const [email, setEmail] = useState(user?.email);
+  const [error, setError] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState(user?.phone_number);
 
   useEffect(() => {
-    getAllTransactions().then((items) => {
-      setTransactions(items);
-    });
+    const callFunc = async () => {
+      if (user != undefined) {
+        const data = await getUser(user?.user_id);
+        setCurrUser(data);
+      }
+    };
+    callFunc();
   }, []);
+
+  const getRole = (head: boolean): string => {
+    if (head) {
+      return "Head";
+    }
+    return "Member";
+  };
+
+  async function onSubmit(event?: React.BaseSyntheticEvent): Promise<void> {
+    event?.preventDefault();
+    const newData = {};
+    try {
+      if (role) {
```

Comment:
I think it may be better to use useState variables here to keep track of user information in the modal (and the default/initial values of the useState variables can be the user's original/pre-edited data)